libs/site-base/src/containers/AddAccount/Generate/index.tsx

```diff
@@ -56,16 +56,19 @@ export const Generate = (props: RouteComponentProps) => {
     const generated = await generateNewWallet(password, setProgress);
     if (generated) {
       const { wallet, decrypted } = generated;
-      accountsApi.addAccount(name, wallet.address, wallet, true, true, decrypted);
+      accountsApi.addAccount(name, wallet.address, wallet);
       registerReferral(wallet.address, referral);
-
-      const toStoragePage = "/addAccount/store" ; //new RUrl(location.pathname, "..", "store");
-      history.push(toStoragePage);
+      setWallet(decrypted);
     }
 
     // Else, we probably cancelled, so do nothing...
     return undefined;
   }
+
+  // Create a new component to finish initialization.  This
+  // is because we cannot add additional hooks to this component to start the account
+  if (wallet)
+    return <CompleteInit signer={wallet} address={wallet.address} redirect={'/addAccount/store'} />
```

Comment:
Icky, but the most react-y way to deal with the need to use a hook to initialize the Account